certbot/display/ops.py

```diff
@@ -289,3 +289,49 @@ def _gen_https_names(domains):
             domains[-1])
 
     return ""
+
+
+def _get_validated(method, validator, message, default=None, **kwargs):
+    if default is not None:
+        try:
+            validator(default)
+        except errors.PluginError as error:
+            raise AssertionError("Invalid default {0}".format(default))
+
+    while True:
+        code, raw = method(message, default=default, **kwargs)
+        if code == display_util.OK:
+            try:
+                validator(raw)
+                return code, raw
+            except errors.PluginError as error:
+                z_util(interfaces.IDisplay).notification(str(error), pause=False)
+        else:
+            return code, raw
+
+
+def validated_input(validator, *args, **kwargs):
+    """Like ~certbot.interfaces.IDisplay.input, but with validation.
```

Comment:
nit: If you surround `~certbot.interfaces.IDisplay.input` with backticks, it gets generated as a link in our documentation which is kind of nice. If you haven't seen it before, here's a [link](https://certbot.eff.org/docs/contributing.html#updating-the-documentation) to how to build our documentation. There's a number of places in this and `validated_directory`'s docstring where this can be done.